contracts/orderbook/Logic/OrderBook.sol

```diff
@@ -0,0 +1,413 @@
+    ) public returns (bool) {
+        uint256 dexRate = getDexRate(
+            swapImpl,
+            base,
+            loanTokenAddress,
+            payload,
+            10**IERC20Metadata(base).decimals()
+        );
+        uint256 indexRate = queryRateReturn(
+            base,
+            loanTokenAddress,
+            10**IERC20Metadata(base).decimals()
+        );
+        if (dexRate >= indexRate) {
+            if (((dexRate - indexRate) * 1000) / dexRate <= 5) {
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            if (((indexRate - dexRate) * 1000) / indexRate <= 5) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+    }
+
+    function executeOrder(bytes32 orderID) external pausable {
+        require(!_allOrders[orderID].isCancelled, "OrderBook: non active");
+        IOrderBook.Order memory order = _allOrders[orderID];
+        address srcToken;
+        uint256 amountUsed;
+        (uint256 dexID, bytes memory payload) = _prepDexAndPayload(order.loanDataBytes);
+        if (payload.length == 0) {
+            if (order.orderType == IOrderBook.OrderType.LIMIT_OPEN) {
+                payload = abi.encode(order.loanTokenAddress, order.base);
+            } else {
+                payload = abi.encode(order.base, order.loanTokenAddress);
+            }
+            dexID = 1;
+        }
+        ISwapsImpl swapImpl = ISwapsImpl(
+            IDexRecords(getSwapAddress()).retrieveDexAddress(dexID)
+        );
+        srcToken = order.collateralTokenAmount > order.loanTokenAmount ? order.base : order.loanTokenAddress;
+        require(order.timeTillExpiration > block.timestamp, "OrderBook: Order Expired");
+        if (_allOrders[orderID].orderType == IOrderBook.OrderType.LIMIT_OPEN) {
+            uint256 dSwapValue;
+            if (srcToken == order.loanTokenAddress) {
+                amountUsed = order.loanTokenAmount + (order.loanTokenAmount * order.leverage) / 10**18; //adjusts leverage
+                (dSwapValue, ) = swapImpl.dexAmountOutFormatted(
+                    payload,
+                    amountUsed
+                );
+            } else {
+                amountUsed = queryRateReturn(
+                    order.base,
+                    order.loanTokenAddress,
+                    order.collateralTokenAmount
+                );
+                amountUsed = (amountUsed * order.leverage) / 10**18;
+                (dSwapValue, ) = swapImpl.dexAmountOutFormatted(
+                    payload,
+                    amountUsed
+                );
+                dSwapValue += order.collateralTokenAmount;
+            }
+
+            require(
+                order.amountReceived <= dSwapValue,
+                "OrderBook: amountOut too low"
+            );
+            _executeTradeOpen(order);
+            _allOrders[orderID].isCancelled = true;
```

Comment:
use `order`

Reply:
Cannot as it is not saving order after so it is easier to write it like this. This is the only spot where things are changed.